Adds Aequatus visualisation plugin

anilthanki opened the pull request:
Adding Aequatus plugin to visualise gene families produced with GeneSeqToFamily workflow.

anilthanki committed:
adds Aequatus visualisation plugin

updates Aequatus control panel-settings font size

Update README.md

fix path to aequatus static scripts

updates with upstream

removes console

removes installation guide

removes demo

removes duplicates

removes d3.js

removes bootstrap

prunes font-awesome

removes less of font-awesome

removes scss of font-awesome

uses font-swesome min

removes bootstrap

removes leftandright css class

updates jQuery and changes css class for popup

updates jQuery

updates fontawesome to 5.6.1

restore galaxy.yml.sample

add missing resource

update paths after updating font

remove console log

switch to minified jquery ui

hexylena commented:
Do you have a screenshot of this? I'm curious what it looks like and if there's an image that's helpful to integrate into usegalaxy.eu's release notes to help our users understand what the new plugin does.

nsoranzo commented:
@erasche It's already installed on usegalaxy.eu :wink:

Some screenshots are in this presentation: https://docs.google.com/presentation/d/1_KdwjbIyjUhdb_huAzOuI34693mciw_vXDCGLctKs9A/edit?usp=sharing

hexylena commented:
Huh. That's. Huh. Wonder how that got there.  Ok, thanks for the pictures! That's super helpful for a viz plugin PR

dannon commented:
@anilthanki I was out last week, but hope to check this out and take a stab at the packaging fix today.  Thanks for the patience!

martenson commented:
@anilthanki @nsoranzo 
I believe the architecture of this plugin (especially dependency handling) does not follow our standards, but given our standards are not well documented, or even defined, the committers have agreed to not fully enforce them. In other words as @jmchilton stated in our discussion:

>"We will merge these in lieu of having a good way to package and distribute these but we are not taking ownership."

If you address the points below I believe this PR will make it to the 19.01 release. It *has to* be re-packaged before 19.05 which we will try to guide you through as much as we can. Otherwise it will be stripped off the codebase.

* loading boostrap using `${h.js('libs/bootstrap'...` will not work since bootstrap is webpacked I believe. The route is returning 404 when I load the plugin, afaik it still seems to work so maybe you do not need that library? Moreover you are including a copy of bootstrap.js in your own static but not loading it.

* on load the viz will request `http://127.0.0.1:8080/plugins/visualizations/aequatus/[object%20Object]` which is an obvious 404


* Please remove the console.logs
```
static/aequatus-vis/scripts/drawCIGARs.js
178:                    console.log(cigar_pos)
179:                    console.log(startposition)
182:                    console.log(startposition)

static/aequatus-vis/scripts/newick.js
94:    console.log(tree)

static/aequatus-vis/scripts/d3_tree.js
42:    console.log("drawTree")
332:    console.log("filterRankUP")
503:    console.log("update")
```

* please remove the duplications if possible

  * aequatus.css is including twice in slightly differing versions
  * prototype.js is included twice
  * control.js is included twice
  * popup.js is including twice with small diff

* other
  * boostrap.min.css seems to be fully commented out
  * do not include `.map` files please
  * prune the fontawesome4.2.0 and bootstrap-css directories, use only the minified versions of the libraries and minimal representation of the font (same with less, scss)
  * remove demo folder
  * remove or adapt readme (it is a carbon copy of your repo readme, which does not apply here)
  * upgrade at least jquery and jquery-ui that have known XSS in the version you included
  * possibly look into updating the rest of dependencies to their latest versions

nsoranzo commented on config/plugins/visualizations/aequatus/README.md:
````diff
@@ -0,0 +1,36 @@
+
+   ```
+   cd GALAXY_DIR/config/plugins/visualizations/
+   ln -s EARLHAM-GALAXYTOOLS_DIR/visualisations/aequatus/
````
You can remove the above installation instructions.

anilthanki commented:
Before starting to fulfil requirements I rebased this branch with Galaxy dev to get all latest code.. 

Hopefully now its ready for release 19.01

nsoranzo commented:
@galaxybot test this

martenson commented:
@anilthanki I am not sure if this is still WIP, but e.g. the jquery library has not been updated and the 404 still occurs

anilthanki commented:
last 3 points were not showing in my email.. I noticed them now..

I will try to get that fix by end of tomorrow

martenson commented:
@anilthanki I added them few minutes later by editing, sorry for the confusion that caused

anilthanki commented:
I have updated jQuery and fontawesome and its using Galaxy's d3 library... 

So from my side its all done..

Also looks like I removed galaxy.yml.sample by mistake in commit "05afe23"

martenson reviewed (approved):
I added bunch more fixes and squash-rebased it to minify the commit.

At this point I am +1 on merging. This is an approval with a hook since it has to be re-packaged before 19.05 as discussed above.

Thanks for the contribution.

anilthanki commented:
Thanks a lot @martenson This is very helpful, yes I agree this needs to be re-packaged before 19.05 please let me know as soon as visualisation documentation is available.

Thanks again

Anil

martenson commented:
@galaxybot test this

dannon commented:
Thanks @anilthanki, and @martenson both for this.
